src/signal.rs

```diff
@@ -474,4 +475,55 @@ mod tests {
         // forever as the loop keeps running. Since we don't join, the thread will become detached
         // as the handle is dropped, and will be killed when the process/main thread exits.
     }
+
+    #[test]
+    fn test_block_unblock_signal() {
+        let signal = SIGRTMIN() + 0;
+
+        // Check if it is blocked.
+        unsafe {
+            let mut sigset: sigset_t = mem::zeroed();
+            pthread_sigmask(SIG_BLOCK, null(), &mut sigset as *mut sigset_t);
+            assert_eq!(sigismember(&sigset, signal), 0);
+        }
+
+        block_signal(signal).unwrap();
+        assert!(get_blocked_signals().unwrap().contains(&(signal)));
+
+        unblock_signal(signal).unwrap();
+        assert!(get_blocked_signals().unwrap().is_empty());
+    }
+
+    #[test]
+    fn test_clear_pending() {
+        let signal = SIGRTMIN() + 1;
+
+        block_signal(signal).unwrap();
+
+        // Block the signal, which means it won't be delivered until it is
+        // unblocked. Pending between the time when the signal which is set as blocked
+        // is generated and when is delivered.
+        let killable = thread::spawn(move || {
+            loop {
+                // Wait for the signal being killed.
+                thread::sleep(Duration::from_millis(100));
+                if unsafe {
+                    let mut chkset: sigset_t = mem::zeroed();
+                    sigpending(&mut chkset);
+                    sigismember(&chkset, signal) == 1
+                } {
+                    clear_signal(signal).unwrap();
+                }
+            }
+        });
+
+        // Send a signal to the thread.
+        assert!(killable.kill(1).is_ok());
+
+        unsafe {
+            let mut chkset: sigset_t = mem::zeroed();
+            sigpending(&mut chkset);
+            assert_eq!(sigismember(&chkset, signal), 0);
```

Comment:
This assertion is always going to be true, because `sigpending` AFAIK returns the pending signals for the current thread, and we only send the signal to `killable`. Is the goal of this test to block a signal, then send it, check that it's pending, clear it, and finally see that it's not pending anymore? If so, can we do this sequence of operations for the current thread, without starting another one with `thread::spawn`?

Reply:
How can we send signal to current thread? :)

Reply:
Refer to http://man7.org/linux/man-pages/man2/sigpending.2.html

Reply:
Yea I think that works as well, and actually makes more sense than what I've said.